### PR TITLE
[VecOps] Add physics helpers

### PR DESCRIPTION
**Pull request by stwunsch**
Implements different physics helpers for RVec collections:
 - `DeltaPhi`: Angle difference with correct handling of boundaries at modulo 2 * pi
 - `DeltaR`: Distance on eta-phi plane
 - `InvariantMass`: Invariant mass of four-vector collections

As follows the output of the tutorial showing the functionality of the helpers.

```bash
DeltaPhi(phi1 = { 0, 0, 0, 1, 1, -0.5 },
         phi2 = { 1, -0.5, 4.14159, -0.5, 4.14159, 4.14159 })
 = { 1, -0.5, -2.14159, -1.5, 3.14159, -1.64159 }

DeltaR(eta1 = { 2.4, 2.4, 2.4, -1.5, -1.5, 1 },
       eta2 = { -1.5, 1, 0, 1, 0, 0 },
       phi1 = { 0, 0, 0, 1, 1, -0.5 },
       phi2 = { 1, -0.5, 4.14159, -0.5, 4.14159, 4.14159 })
 = { 4.02616, 1.48661, 3.21659, 2.91548, 3.48132, 1.92219 }

InvariantMass(pt1 = { 40, 20, 30 },
              eta1 = { 2.5, 0.5, -1 },
              phi1 = { -0.5, 0, 1 },
              mass1 = { 10, 10, 10 },
              pt2 = { 20, 10, 40 },
              eta2 = { 0.5, -0.5, 1 },
              phi2 = { 0, 1, -1 },
              mass2 = { 2, 2, 2 })
 = { 69.0799, 23.6971, 101.326 }

InvariantMass(pt = { 40, 20, 30 },
              eta = { 2.5, 0.5, -1 },
              phi = { -0.5, 0, 1 },
              mass = { 10, 10, 10 })
 = 220.308
```

### CODE REVIEW

File: math/vecops/inc/ROOT/RVec.hxx [hunk 1]
```diff
@@ -1242,6 +1242,52 @@ RVec<T> Where(const RVec<int>& c, T v1, T v2)
    return r;
 }
 
+/// Return the angle difference (delta phi) of two scalars.
+/// The function computes the closest angle from v1 to v2 with sign and is
+/// therefore in the range [-M_PI, M_PI].
```
Comment (dpiparo) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// therefore in the range \f$[-\pi, \pi]\f$.
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 2]
```diff
@@ -1242,6 +1242,52 @@ RVec<T> Where(const RVec<int>& c, T v1, T v2)
...
+/// The computation is done per default in radians (c == M_PI) but can be switched
```
Comment (dpiparo) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// The computation is done per default in radians \f$c = \pi\f$ but can be switched
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 3]
```diff
@@ -1242,6 +1242,52 @@ RVec<T> Where(const RVec<int>& c, T v1, T v2)
    return r;
 }
 
+/// Return the angle difference (delta phi) of two scalars.
```
Comment (dpiparo) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// Return the angle difference \f$\Delta \phi\f$ of two scalars.
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 4]
```diff
@@ -1242,6 +1242,52 @@ RVec<T> Where(const RVec<int>& c, T v1, T v2)
...
+/// The function computes the closest angle from v1 to v2 with sign and is
+/// therefore in the range [-M_PI, M_PI].
+/// The computation is done per default in radians (c == M_PI) but can be switched
+/// to degrees (c == 180).
```
Comment (dpiparo) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// to degrees \f$c = 180\f$.
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 5]
```diff
@@ -1242,6 +1242,52 @@ RVec<T> Where(const RVec<int>& c, T v1, T v2)
...
+   if (r < -c) {
+      r += 2.0 * c;
+   }
+   else if (r > c) {
+      r -= 2.0 * c;
+   }
+   return r;
+}
+
+/// Return the angle difference (delta phi) in radians of two vectors.
```
Comment (dpiparo) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// Return the angle difference \f$\Delta \phi\f$ in radians of two vectors.
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 6]
```diff
@@ -1242,6 +1242,52 @@ RVec<T> Where(const RVec<int>& c, T v1, T v2)
...
+   using size_type = typename RVec<T>::size_type;
+   const size_type size = v1.size();
+   auto r = RVec<T>(size);
+   for (size_type i = 0; i < size; i++) {
+      r[i] = DeltaPhi(v1[i], v2[i], c);
+   }
+   return r;
+}
+
+/// Return the angle difference (delta phi) in radians of a vector and a scalar.
```
Comment (dpiparo) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// Return the angle difference \f$\Delta \phi\f$ in radians of a vector and a scalar.
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 7]
```diff
@@ -1242,6 +1242,52 @@ RVec<T> Where(const RVec<int>& c, T v1, T v2)
...
+template <typename T>
+RVec<T> DeltaPhi(const RVec<T>& v1, T v2, const T c = M_PI)
```
Comment (dpiparo) [inline math/vecops/inc/ROOT/RVec.hxx]: Do we need also a simple forwarder with the signature `RVec<T> DeltaPhi(T, const RVec<T>& , const T)`

File: tutorials/vecops/vo007_PhysicsHelpers.C [hunk 8]
```diff
@@ -0,0 +1,35 @@
+/// \file
+/// \ingroup tutorial_vecops
+/// \notebook -nodraw
+/// In this tutorial we demonstrate RVec helpers for physics computations such
+/// as angle differences (delta phi).
```
Comment (dpiparo) [inline tutorials/vecops/vo007_PhysicsHelpers.C]:
```suggestion
/// as angle differences \f$\Delta \phi\f$.
```

File: math/vecops/inc/ROOT/RVec.hxx (hunk 7 above)
Comment (stwunsch) [reply math/vecops/inc/ROOT/RVec.hxx]: I put it in!

**Comment by dpiparo** [comment]
As soon as the PR builds and passes all tests, it can be merged.

**Comment by stwunsch** [comment]
I'll rebuild later. `M_PI` should now be there on windows following [this](https://docs.microsoft.com/en-us/cpp/c-runtime-library/math-constants?view=vs-2019) doc.

**Comment by stwunsch** [comment]
@phsft-bot build on `windows10/default`

**Pull request merged**